docs/dev/project-structure.md

```diff
@@ -0,0 +1,46 @@
+# Project structure
+
+If you're not familiar with the overall module structure from the SDK modules, please check this [document](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) as prerequisite reading.
+
+Every Interchain Standard (ICS) has been developed in its own package. The development team separated the IBC TAO (Transport, Authentication, Ordering) ICS specifications from the IBC application level specification. The following sections describe the architecture of the most relevant directories that comprise this repository.
+
+## `modules` 
+
+This folder contains implementations for the IBC TAO (`core`), IBC applications (`apps`) and light clients (`light-clients`). 
+
+### `core`
+
+- `02-client`: This package is an implementation for Cosmos SDK-based chains of [ICS-02](https://github.com/cosmos/ibc/tree/main/spec/core/ics-002-client-semantics). This implementation defines the types and methods needed to operate light clients tracking other chain's consensus state.
+- `03-connection`: This package is an implementation for Cosmos SDK-based chains of [ICS-03](https://github.com/cosmos/ibc/tree/main/spec/core/ics-003-connection-semantics). This implementation defines the types and methods necessary to perform connection handshake between two chains.
+- `04-channel`: This package is an implementation for Cosmos SDK-based chains of [ICS-04](https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics). This implementation defines the types and methods necessary to perform channel handshake between two chains and ensure correct packet sending flow.
+- `05-port`: This package is an implementation for Cosmos SDK-based chains of [ICS-05](https://github.com/cosmos/ibc/tree/main/spec/core/ics-005-port-allocation). This implements the port allocation system by which modules can bind to uniquely named ports.
+- `23-commitment`: This package is an implementation for Cosmos SDK-based chains of [ICS-23](https://github.com/cosmos/ibc/tree/main/spec/core/ics-023-vector-commitments). This implementation defines the functions required to prove inclusion or non-inclusion of particular values at particular paths in state.
+- `24-host`: This package is an implementation for Cosmos SDK-based chains of [ICS-24](https://github.com/cosmos/ibc/tree/main/spec/core/ics-024-host-requirements).
+
+### `apps`
+
+- `transfer`: This is the Cosmos SDK implementation of the [ICS-20](https://github.com/cosmos/ibc/tree/main/spec/app/ics-020-fungible-token-transfer) protocol, which enables cross-chain fungible token transfers. For more information, read the [module's docs](../apps/transfer/overview.md)
```

Comment:
I think we should rename this one to `20-transfer`, it seems to be the odd one out at the moment! (not in this PR of course!)

Reply:
I agree. Something to do at some point!